### PR TITLE
Updating extension guidelines v2

### PR DESCRIPTION
**Pull request by untitaker**
Continuing #1035 and #1218, #1135 related.

### CODE REVIEW

**Comment by untitaker** [comment]
This is a complete mess and doesn't merge anymore, we should cherry-pick out of it and kill it then.

File: docs/extensiondev.rst [hunk 1]
```diff
@@ -410,8 +404,8 @@ deprecated ``flaskext.foo``.  Flask 0.8 introduces a redirect import
 system that lets uses import from ``flask.ext.foo`` and it will try
```
Comment (harrymvr) [inline docs/extensiondev.rst]: There's a typo here. `uses` -> `users`

**Comment by jeffwidman** [comment]
Taken care of in #1484

**Comment by untitaker** [comment]
Not really, we still need to update the extension guidelines in some way.

**Comment by jeffwidman** [comment]
My bad. 

Can you clarify what's missing? 

005a49d969e2e4ebcd23cfd8347a6f2d331e1778 updated the extension guidelines and 188f0e2710bec195925687f51db5dc430ee5ee91 added a deprecation warning. 

Is there something else beyond the switch to `flask_extensionname`?

**Comment by untitaker** [comment]
@jeffwidman Yes, I think the enumerated list for extension requirements has to be updated as well. That is a completely unrelated change, see #1035

**Comment by kennethreitz** [comment]
needs a rebase!

**Comment by kennethreitz** [comment]
i merged this manually.

**Pull request merged**